lib/IDE/CompletionInstance.cpp

```diff
@@ -155,10 +155,7 @@ bool CompletionInstance::performCachedOperaitonIfPossible(
     return false;
 
   auto &oldInfo = oldState.getCodeCompletionDelayedDeclState();
-
-  // Currently, only completions within a function body is supported.
-  if (oldInfo.Kind != CodeCompletionDelayedDeclKind::FunctionBody)
```

Comment:
Is it worth doing this check later inside the switch when `newInfo.Kind == FunctionBody`?  Or is it covered by the check of `getEquivalentDeclContextFromSourceFile`

Reply:
I believe it covered by interface hash check. If the previous completion happened in toplevel or type members, it must changes the interface hash.